### PR TITLE
Adds a filter to the passive analysis for stopwords and stop characters without preceding space

### LINKED ISSUE #962: Finetune English Passive voice detection
- As far as I know, 'so-called' is already included in the exception list. However, it is still marked as a passive. 
- 'Related', 'red', 'based' and 'located' should be added to the non-verb -ed exception list
- These additional stopwords should be added: “and", “but”, “or”

### LINKED ISSUE #1223: Passive voice analysis: stop words not recognized when there is no preceding space
The following stop words in the passive voice analysis are not recognised because they are not preceded by a space:
- ‘ll (such as in “he’ll”)
- :

In addition, the following characters should be added as stop words once this problem is solved: 

- ,
- –

### PR DESCRIPTION
**Pull request by manuelaugustin**
## Summary

This PR can be summarized in the following changelog entry: Adds a filter to avoid incorrectly marking participles as passive when there is a stopword or relevant puncuation mark between the auxiliary and the passive that isn't preceded by a space. For example, "It's a hands-free, voice-controlled device" will not be marked as passive because there is a comma between the auxiliary "it's" and the -ed word "voice-controlled". Also adds some additional stopwords (“–“, “and”, “or”, “but”) and non-verbs ending in -ed exceptions ("based”, “located”, “red”, “related”).

## Test instructions

This PR can be tested by following these steps:

- Use the browserified example. Don't forget to run `grunt build:js`.
- Paste a passive English sentence (e.g., `I was helped`)
- Add a string including a stop word (e.g., `'ll` and `'ve` such as in `she'll` and `we've`) or relevant punctuation marks (`:` and `,`) between `was` and `helped`.
- To test the new stopwords (surrounded by spaces), add one of these words between `was` and `helped`.
- To test the new non-verbs ending in -ed exceptions, replace `helped` with one of these words.
- Make sure none of these sentences are marked as passive

Fixes #1223
Fixes #962

### CODE REVIEW

File: spec/researches/english/getSentencePartsSpec.js [hunk 1]
```diff
@@ -17,4 +17,14 @@ describe( "splits English sentences into parts", function() {
 		expect( getSentenceParts( sentence )[ 0 ].getSentencePartText() ).toBe( "is praise due." );
 		expect( getSentenceParts( sentence ).length ).toBe( 1 );
 	} );
+	it ( "splits sentences on stop characters", function() {
+		var sentence = "It is a hands-free, voice-controlled device.";
+		expect( getSentenceParts( sentence )[ 0 ].getSentencePartText() ).toBe( "is a hands-free" );
+		expect( getSentenceParts( sentence ).length ).toBe( 1 );
+		} );
+	it ( "doesn't sentences on stop characters that are not preceded by a word and followed by a space/punctuation mark", function() {
```
Comment (Xyfi) [inline spec/researches/english/getSentencePartsSpec.js]: The test description is not a valid sentence.

File: spec/researches/english/getSentencePartsSpec.js [hunk 2]
```diff
@@ -17,4 +17,24 @@ describe( "splits English sentences into parts", function() {
 		expect( getSentenceParts( sentence )[ 0 ].getSentencePartText() ).toBe( "is praise due." );
 		expect( getSentenceParts( sentence ).length ).toBe( 1 );
 	} );
+	it ( "splits sentences on stop characters", function() {
+		var sentence = "It is a hands-free, voice-controlled device.";
+		expect( getSentenceParts( sentence )[ 0 ].getSentencePartText() ).toBe( "is a hands-free" );
+		expect( getSentenceParts( sentence ).length ).toBe( 1 );
+		} );
+	it ( "doesn't split sentences on stop characters that are not preceded by a word and followed by a space/punctuation mark", function() {
```
Comment (IreneStr) [inline spec/researches/english/getSentencePartsSpec.js]: Doesn't the next test also test this?

File: src/.baseDir.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+// Ignore this file. See https://github.com/grunt-ts/grunt-ts/issues/77
```
Comment (IreneStr) [inline src/.baseDir.ts]: Please remove this file from the commit

File: src/researches/english/passivevoice/non-verb-ending-ed.js [hunk 4]
```diff
@@ -76,6 +76,7 @@ module.exports = function() {
 		"barefooted",
 		"barelegged",
 		"barnacled",
+		"based",
```
Comment (IreneStr) [inline src/researches/english/passivevoice/non-verb-ending-ed.js]: Please mention these additions in your PR message as well.

File: src/researches/english/passivevoice/stopwords.js [hunk 5]
```diff
@@ -43,7 +43,9 @@ module.exports = function() {
 		"won't",
 		"do",
 		"does",
-		"'ll",
-		":",
+		"–",
+		"and",
```
Comment (IreneStr) [inline src/researches/english/passivevoice/stopwords.js]: Please mention these additions in your PR message as well.

File: spec/stringProcessing/indicesSpec.js [hunk 6]
```diff
@@ -4,6 +4,7 @@ let getIndicesOfList = indices.getIndicesByWordList;
 let filterIndices = indices.filterIndices;
 let sortIndices = indices.sortIndices;
 let getIndicesByWord = indices.getIndicesByWord;
+let getIndicesByStopCharacterList = indices.getIndicesByStopCharacterList;
```
Comment (IreneStr) [inline spec/stringProcessing/indicesSpec.js]: Staat niet in de export van indices.js.
+ naming isn't really clear.

**Comment by IreneStr** [comment]
CR done 👍  I've clarified the descriptions of one of the tests a bit.

**Comment by IreneStr** [comment]
Acceptance done 👍

**Pull request merged**